Dynatrace/dynatrace-operator
# EdgeConnect failing because v1alpha1 converter to v1alpha2 deson't handle properly empty .spec.HostRestrictions

## Description

This PR fixes bug https://dt-rnd.atlassian.net/browse/K8S-11404

when we don't provide .spec.HostRestrictions in EdgeConnect CR (default case, it's not required)

and we render wrong config for EdgeConnect pod, during converting v1alpha1 to v1alpha2.

## How can this be tested?

1. Create EdgeConnect inside your tenant
2. update and apply CR without hostRestrictions:

```
piVersion: dynatrace.com/v1alpha1
kind: EdgeConnect
metadata:
  name: e2e-andrii
  namespace: dynatrace
spec:
  apiServer: "mts86258.dev.apps.dynatracelabs.com"
  replicas: 1
  oauth:
    clientSecret: edge-connect-client-secret-mts86258
    endpoint: https://sso-dev.dynatracelabs.com/sso/oauth2/token
    resource: urn:dtenvironment:mts86258
---
apiVersion: v1
kind: Secret
metadata:
  name: edge-connect-client-secret-mts86258
  namespace: dynatrace
data:
  oauth-client-id: ***
  oauth-client-secret: ****
```

3. Check edgeconnect pod, it must run no errors, 

Bonus: you can apply another EC with .spec.Hostrestrictions.
